src/basic/sysctl-util.c

```diff
@@ -58,6 +58,28 @@ int sysctl_write(const char *property, const char *value) {
 
         log_debug("Setting '%s' to '%s'", p, value);
 
+        if (shadow) {
+                char *k, *v;
+                int r;
+
+                k = strdup(p);
+                if (!k)
+                        return -ENOMEM;
+
+                v = strdup(value);
+                if (!v) {
+                        free(k);
+                        return -ENOMEM;
+                }
+
+                r = hashmap_ensure_replace(shadow, &path_hash_ops_free_free, k, v);
+                if (r < 0) {
+                        free(k);
+                        free(v);
```

Comment:
as mentioned, please use `_cleanup_free_` when declaring these variables, so that they are freed automatically, when we leave scope.

And then, when the hashmap_ensure_replace() actually worked successfully, please add `TAKE_PTR(k);` and `TAKE_PTR(v);` to make sure the variables are reset to NULL so that the auto-destruction is disabled.